bamboo_twig_security/src/TwigExtension/Security.php

```diff
@@ -83,4 +132,52 @@ public function hasRole($role, $user = NULL) {
     return $account->hasRole($role);
   }
 
+  /**
+   * Does the current|given user has the given roles collection ?
+   *
+   * @param string[] $roles
+   *   Drupal roles name.
+   * @param string $conjunction
+   *   (Optional) The conjunction to use againts user permissions.
+   *   Allowing 'AND' or 'OR' values. When nothing is given 'AND' is used.
+   * @param int $user
+   *   (Optional) user id to check permission. Otherwise current user is used.
+   *
+   * @return bool
+   *   True if the current|given user has the given permission. Otherwise FALSE.
+   */
+  public function hasRoles($roles, $conjunction = 'AND', $user = NULL) {
+    // Sanitize the conjunction to AND / OR values.
+    if (!in_array($conjunction, ['AND', 'OR'])) {
+      $conjunction = 'AND';
+    }
+
+    $currentUser = $this->getCurrentUser();
+    if (is_null($user) && $currentUser->isAnonymous()) {
+      return NULL;
+    }
+
+    $user_id = $currentUser->id();
+    if (!is_null($user) && is_int($user)) {
+      $user_id = $user;
+    }
+
+    $account = $this->getUserStorage()->load($user_id);
+    if (!$account) {
+      return NULL;
+    }
+
+    foreach ($roles as $role) {
+      if ($conjunction == 'OR' AND $account->hasRole($role) ) {
+        return TRUE;
+      }
+
+      if ($conjunction == 'AND' AND !$account->hasRole($role) ) {
+        return FALSE;
+      }
+    }
+
+    return TRUE;
```

Comment:
Your are right I should

```
$conjunction == 'OR' ? FALSE : TRUE;
```

Because when 'AND' Conjunction you will never return in the loop if you have all & the conjunction 'OR' will return FALSE if not return on the loop.